crates/relayer/src/config.rs

```diff
@@ -556,14 +558,37 @@ pub enum EventSourceMode {
 
 #[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
 #[serde(deny_unknown_fields)]
-pub struct ChainConfig {
+#[serde(tag = "type")]
+pub enum ChainConfig {
+    CosmosSdk(CosmosSdkConfig),
+}
+
+impl ChainConfig {
+    pub fn id(&self) -> &ChainId {
+        match self {
+            Self::CosmosSdk(config) => &config.id,
+        }
+    }
+
+    pub fn packet_filter(&self) -> &PacketFilter {
+        match self {
+            Self::CosmosSdk(config) => &config.packet_filter,
+        }
+    }
+
+    pub fn max_block_time(&self) -> Duration {
+        match self {
+            Self::CosmosSdk(config) => config.max_block_time,
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct CosmosSdkConfig {
```

Comment:
This should move to the `cosmos/` backend but I left it for now for ease of initial commentary/review.